src/utilities.jl

```diff
@@ -221,14 +221,14 @@ Build a VRT from a list of datasets.
 The output dataset.
 """
 function unsafe_gdalbuildvrt(
-        datasets::Vector{Dataset},
+        datasets::Vector{T},
         options = String[];
         dest = "/vsimem/tmp"
-    )
+    ) where T <:AbstractDataset
     options = GDAL.gdalbuildvrtoptionsnew(options, C_NULL)
     usage_error = Ref{Cint}()
     result = GDAL.gdalbuildvrt(dest, length(datasets),
         [ds.ptr for ds in datasets], C_NULL, options, usage_error)
     GDAL.gdalbuildvrtoptionsfree(options)
-    return Dataset(result)
+    return T(result)
```

Comment:
This should still be `Dataset(result)` rather than `T(result)` (see https://github.com/yeesian/ArchGDAL.jl/pull/167#issuecomment-765704084 for context).